### ISSUE TITLE
elastic-apm-http-client@9.7.0 is currently breaking tests

### ISSUE THREAD

**Issue by trentm** [elastic/apm-agent-nodejs#2032]
I released elastic-apm-http-client@9.7.0 earlier today with the "blocking behaviour" fix in prep for #2024 *using* it. The http client changes were significant enough that there are test changes necessary in apm-agent-nodejs. (AFAIK currently *just* test changes are required.)

The current state is breaking tests for this repo because the current dependency spec is:

```
"elastic-apm-http-client": "^9.5.1",
```

which is picking up the 9.7.0 release.

To **workaround** this for now, I'd like to do a apm client release that includes locks  to using the previous version of the http client until #2024 is ready.

* * *

Update: v9.7.0 of the client *does* introduce a subtle change in 'uncaughtException' handling behaviour. See https://github.com/elastic/apm-nodejs-http-client/issues/150

**Comment by trentm** [comment]
An alternative to consider is `npm unpublish`ing this specific http client version. https://www.npmjs.com/policies/unpublish policy allows that within 72 hours. THis would mean that users won't need to upgrade their apm client version to get the "fix".

This should only need to be considered if we think there is a change the http-client release is breaking behaviour.

The tests say it isn't... but I'm not 100% confident because I think I'm potentially seeing a change in handling the exit status for our uncaughtException handling in my side-work to convert from tape to tap. :/ I haven't yet confirmed that.

**Comment by trentm** [comment]
I think the best answer is to unpublish this version. We'll need to rerelease it as a subsequent version, but this is by far the cleanest way to avoid this bug.

A timing issue is that I want to do it *right now* because we can only unpublish if "had less than 300 downloads over the last week". Currently that is true:

![Screen Shot 2021-04-05 at 7 05 13 PM](https://user-images.githubusercontent.com/46866/113648708-e5f4d300-9641-11eb-823b-b912d7c66db9.png)

but I think if left overnight that might no longer be true.

I've tested unpublishing a version on another old unused package of mine to confirm:

```
[19:01:13 trentm@pink:~/tm/node-kthxbai (git:master rv:2)]
% npm publish
npm notice
npm notice 📦  kthxbai@0.4.2
...
+ kthxbai@0.4.2

% npm unpublish kthxbai@0.4.2
...
- kthxbai@0.4.2
```

The latest version is now 0.4.0 (the previously published version), and 0.4.2 still sits there:

```
% npm info -j
{
  "_id": "kthxbai@0.4.0",
  "_rev": "8-cc7fc9170dca0380b34552a09f80552e",
  "name": "kthxbai",
  "description": "Kill Those Huge eXtra Bits Absentmindedly Installed -- trim files from node_modules not needed for running",
  "dist-tags": {
    "latest": "0.4.0"
  },
  "versions": [
    "0.3.0",
    "0.3.1",
    "0.4.0"
  ],
  "maintainers": [
    "trentm <trentm@gmail.com>"
  ],
  "time": {
    "modified": "2021-04-06T02:02:08.016Z",
    "created": "2015-01-05T05:31:22.096Z",
    "0.3.0": "2015-01-05T05:31:22.096Z",
    "0.3.1": "2015-01-05T22:17:58.201Z",
    "0.4.0": "2015-01-27T02:01:58.332Z",
    "0.4.2": "2021-04-06T02:01:26.023Z"
  },
```

An install of latest gets 0.4.0:

```
% npm install kthxbai
...
+ kthxbai@0.4.0
added 20 packages from 12 contributors and audited 20 packages in 1.317s
...
```

And direct attempt to get the unpublished version fails:

```
% npm install kthxbai@0.4.2
npm ERR! code ETARGET
npm ERR! notarget No matching version found for kthxbai@0.4.2.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
...
```

**Comment by trentm** [comment]
@astorm FYI, I'm unpublishing this one now.

**Comment by trentm** [comment]
Nevermind, cannot unpublish. I guess I misinterpreted the "no other packages in the npm Public Registry depend on"

```
% npm unpublish elastic-apm-http-client@9.7.0
This operation requires a one-time password.
Enter OTP: 177770
npm ERR! code E405
npm ERR! 405 Method Not Allowed - PUT https://registry.npmjs.org/elastic-apm-http-client/-rev/47-33737dd71a1c7f3989aae381f5f568c9 - You can no longer unpublish this package.
npm ERR! Failed criteria:
npm ERR! has dependent packages in the registry
npm ERR!
npm ERR! Please deprecate it instead:
npm ERR! npm deprecate -f 'elastic-apm-http-client@9.7.0' "this package has been deprecated"
npm ERR! To learn more about our unpublish policies, see https://www.npmjs.com/policies/unpublish

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/trentm/.npm/_logs/2021-04-06T02_10_25_726Z-debug.log
```

Back to considering a release of the agent locking to "9.6.0" of the http client. I'll wait on this until tomorrow.
